src/network.c

```diff
@@ -643,32 +647,47 @@ static int write_part_values (char **ret_buffer, int *ret_buffer_len,
 	for (i = 0; i < num_values; i++)
 	{
 		pkg_values_types[i] = (uint8_t) ds->ds[i].type;
-		switch (ds->ds[i].type)
-		{
-			case DS_TYPE_COUNTER:
-				pkg_values[i].counter = htonll (vl->values[i].counter);
-				break;
-
-			case DS_TYPE_GAUGE:
-				pkg_values[i].gauge = htond (vl->values[i].gauge);
-				break;
-
-			case DS_TYPE_DERIVE:
-				pkg_values[i].derive = htonll (vl->values[i].derive);
-				break;
 
-			case DS_TYPE_ABSOLUTE:
-				pkg_values[i].absolute = htonll (vl->values[i].absolute);
-				break;
-
-			default:
-				free (pkg_values_types);
-				free (pkg_values);
-				ERROR ("network plugin: write_part_values: "
-						"Unknown data source type: %i",
-						ds->ds[i].type);
+		if (ds->ds[i].type == DS_TYPE_GAUGE)
+		{
+			pkg_values[i].gauge = htond (vl->values[i].gauge);
```

Comment:
Nitpick: by convention, blocks containing only one statement don't take brackets (like you did on line 658 for example).

Could you please correct this one as well as those below ? Thanks !